src/features/content-explorer/content-explorer/ContentExplorer.js

```diff
@@ -508,12 +578,23 @@ class ContentExplorer extends Component {
                 >
                     {headerActionsAccessory}
                 </ContentExplorerHeaderActions>
-                {isSelectAllAllowed && (
-                    <ContentExplorerSelectAll
-                        numTotalItems={numTotalItems}
+                {onIncludeSubfoldersToggle && isSelectAllAllowed ? (
+                    <ContentExplorerIncludeSubfolders
+                        hideSelectAllCheckbox={isIncludeSubfoldersEnabled}
                         isSelectAllChecked={isSelectAllChecked}
-                        handleSelectAllClick={this.handleSelectAllClick}
+                        onIncludeSubfoldersToggle={this.handleIncludeSubfoldersToggle}
+                        onSelectAllClick={this.handleSelectAllClick}
+                        isToggleDisabled={this.includeSubfolderToggleDisabled()}
+                        tooltipMessageForToggle={tooltipMessageForToggle}
                     />
+                ) : (
+                    isSelectAllAllowed && (
+                        <ContentExplorerSelectAll
+                            numTotalItems={numTotalItems}
+                            isSelectAllChecked={isSelectAllChecked}
+                            handleSelectAllClick={this.handleSelectAllClick}
+                        />
+                    )
```

Comment:
this is how I think we can combine the checkboxes used for both:

```
{includeSubfoldersProps && <ContentExplorerIncludeSubfolders {...includeSubfoldersProps} />
{isSelectAllAllowed &&
        <ContentExplorerSelectAll
                handleSelectAllClick={this.handleSelectAllClick}
                isLabelHidden={includeSubfoldersProps}
                isSelectAllChecked={isSelectAllChecked}
                numTotalItems={numTotalItems}
        />
```

then inside of `ContentExplorerSelectAll`:
```
const ContentExplorerSelectAll = ({ intl, numTotalItems = 0, isLabelHidden = false, isSelectAllChecked, handleSelectAllClick }) => (
    <div className="content-explorer-select-all-container">
        {!isLabelHidden && <label className="content-explorer-select-all-items-counter">
            {numTotalItems === 1 ? (
                <FormattedMessage {...messages.result} values={{ itemsCount: intl.formatNumber(numTotalItems) }} />
            ) : (
                <FormattedMessage {...messages.results} values={{ itemsCount: intl.formatNumber(numTotalItems) }} />
            )}
        </label>
        }
        {!isLabelHidden && <label className="content-explorer-select-all-checkbox-label">
            <FormattedMessage {...messages.selectAll} />
        </label>
        }
        <Checkbox
            hideLabel
            className="content-explorer-select-all-checkbox"
            onChange={handleSelectAllClick}
            isChecked={isSelectAllChecked}
        />
    </div>
);
```

Reply:
In addition to this we would also have to wrap the Checkbox in a conditional since it is supposed to be hidden when the toggle is turned on. Also, how can we render the tooltip given that it is only supposed to show when we have the toggle

Reply:
whoops forgot to add the tooltip:

const ContentExplorerSelectAll = ({ intl, numTotalItems = 0, isLabelHidden = false, isSelectAllChecked, handleSelectAllClick }) => (
    <div className="content-explorer-select-all-container">
        {!isLabelHidden && <label className="content-explorer-select-all-items-counter">
            {numTotalItems === 1 ? (
                <FormattedMessage {...messages.result} values={{ itemsCount: intl.formatNumber(numTotalItems) }} />
            ) : (
                <FormattedMessage {...messages.results} values={{ itemsCount: intl.formatNumber(numTotalItems) }} />
            )}
        </label>
        }
        {!isLabelHidden && <label className="content-explorer-select-all-checkbox-label">
            <FormattedMessage {...messages.selectAll} />
        </label>
        }
        <Tooltip isShown={isLabelHidden ? undefined : false}> // undefined will give it the default hover behavior. false will hide it
            <Checkbox
                hideLabel
                className="content-explorer-select-all-checkbox"
                onChange={handleSelectAllClick}
                isChecked={isSelectAllChecked}
            />
        </Tooltip>
    </div>
);

Reply:
the parent component should know if the toggle is enabled or disabled and can change value of prop `isSelectAllAllowed`. this way the two features are kept independent of each other

Reply:
Alright, I made all of the react changes that you mentioned here for it to work while still being independent. It does work, the only issue is that  the two components have different css and without changing any of the css, the modal looks like this: 
![Screen Shot 2023-02-16 at 2 52 18 PM](https://user-images.githubusercontent.com/56702540/219505913-91e31be0-d78d-401b-a1e6-a599a9360d1c.png)

Reply:
This is because the containers for the ContentExplorerIncludeSubfodlers component has a flex layout and the ContentExplorerSelectAll component has a display block, each with their own respective heights. Should I mess around with the  original ContennExplorerSelectAll css? Or should I try to make some new css classnames and have them conditionally set on the checkbox based on the isLabelHidden prop?

Reply:
I am leaning towards the latter at the moment

Reply:
![Screen Shot 2023-02-16 at 3 14 50 PM](https://user-images.githubusercontent.com/56702540/219509301-2ff941b0-11a8-46ed-af67-782aff4148a1.png)
![Screen Shot 2023-02-16 at 3 15 16 PM](https://user-images.githubusercontent.com/56702540/219509365-ea351aba-f48f-4cbd-93e3-4dceb5cf17f7.png)



 Got it looking like this at the moment